### PR TITLE
Refactor: Update getimagesize() Function to Accept Image Paths Instead of URLs

### LINKED ISSUE #187: Warning: getimagesize: failed to open stream
When installing this plugin in a local WordPress 5.8 development environment, the following error is displayed when rendering the license:

> Warning: getimagesize(http://localhost:8888/wp-content/plugins/wp-plugin-creativecommons/includes/images/by.png): failed to open stream: Cannot assign requested address in /var/www/html/wp-content/plugins/wp-plugin-creativecommons/includes/class-creativecommons.php on line 1128 

## Steps to reproduce

1. set up WordPress 5.8 (locally or on a server) in development mode
2. install the latest version of the code from this repository in the WordPress plugins directory
3. Go to Settings -> Creative Commons
4. configure the desired license
5. make sure to select "Display license as:" "widget" 
6. go to Appearance -> Widgets
7. add a "Legacy Widget" and select "CC License" widget
8. save the widget settings
9. view the site and notice the Warning in the widget area

## Task
- [ ] reproduce the issue as described above
- [ ] determine the severity of the notices 
- [ ] if needed, fix fhe issue so the notices no longer display

### PR DESCRIPTION
**Pull request by ArchanRD**
## Fixes

Fixes #187 by @brylie

## Description

The function ` getimagesize() ` takes image as parameter and returns the image size i.e height and width. In this PR, we replace image url with image path to fix this issue.

## Technical details

This issue doesn't occur everytime, however in some cases when a website: 
- Has basic auth
- Timesout while loading the image

the image may not be accessible through the image url, hence calling the ` getimagesize() ` with image url may not work always. Also everytime ` getimagesize() ` function is called, the server sends a request to load the image increasing execution time. Hence I've chosen to pass image path instead of image url in ` getimagesize() `. This solution handles the above 2 cases as well as addresses the performace issue of the code.

## Tests

As described above, this situation doesn't always occur, so it may be difficult to test.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by possumbilities** [comment]
@ArchanRD Maybe I'm missing something, but why not make the change further up the chain of functionality like here, so that the `image_url` when passed to the function later behaves normally rather than relying on the function to further modify/cleanup the url?
* https://github.com/creativecommons/wp-plugin-creativecommons/blob/2871fd257a642d75dcc23054949f71787954d4ac/includes/class-creativecommons.php#L1061C4-L1061C50

**Comment by ArchanRD** [comment]
I have simplified the implementation a bit, so it's essentially a one liner change ( but GH is showing it a bit differently though ).

File: includes/class-creativecommons.php [hunk 1]
```diff
@@ -1125,10 +1125,11 @@ public function license_html_rdfa( $deed_url, $license_name, $image_url,
 									$title_work, $is_singular, $attribute_url,
 									$attribute_text, $source_work_url,
 									$extra_permissions_url, $additional_attribution_txt, $title, $title_url, $author, $author_url ) {
-		list($img_width, $img_height) = getimagesize( $image_url );
-		$lazy = function_exists('wp_lazy_loading_enabled') && wp_lazy_loading_enabled('img', 'license_html_rdfa');
+		$image_path = str_replace( CCPLUGIN__URL, CCPLUGIN__DIR, $image_url );
```
Comment (possumbilities) [inline includes/class-creativecommons.php:1128]: I believe the two global variables here should be properly escaped. the value of $image_url is already escaped in its definition further up in the file, but the use of these two globals here doesn't provide them as escaped values.

**Review by possumbilities — CHANGES_REQUESTED** [review]

**Comment by ArchanRD** [comment]
Hey @possumbilities! Sorry I took a lot time to get back to this. I have properly escaped the global variables using `esc_attr()`. Please have a loot at it.

**Comment by possumbilities** [comment]
@ArchanRD Wonderful, thank you! I'll move to testing this one more time and if all is well we'll move it to Approved!

**Review by possumbilities — APPROVED** [review]

**Pull request merged**